code/datums/emotes.dm

```diff
@@ -0,0 +1,152 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+/datum/emote
+	var/key = "" //What calls the emote
+	var/key_third_person = "" //This will also call the emote
+	var/message = "" //Message displayed when emote is used
+	var/message_param = "" //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/muzzled_noise = "" // For snowflaking the muzzled noise
+	var/list/mob_type_blacklist_typecache //Types that are NOT allowed to use that emote
+	var/punct = "."
+	var/list/mob_type_ignore_stat_typecache
+	var/stat_allowed = CONSCIOUS
+	var/cooldown = 0
+	var/sound
+	var/sound_volume = 50
+	var/sound_vary = 0
+	var/static/list/emote_lists = list(
+		mob=init_subtypes(/datum/emote/mob),
+		living=init_subtypes(/datum/emote/living),
```

Comment:
Also, why does this even exist? If you are going to end up only initializing 1 of each emote just do it in a global list and reference that

Reply:
Add a second global list and define a `var/sorting_key` on /datum/emote to make a list like this e-z p-z

Reply:
MrStonedOne says static vars are better for performance http://www.byond.com/forum/?post=2140538

Also the original tg implementation also used a static variable